File: onyxsdk-base/src/main/java/com/onyx/android/sdk/data/Size.java
package com.onyx.android.sdk.data;

import android.graphics.RectF;
import androidx.annotation.NonNull;
import java.util.Objects;

public class Size {
   public int width;
   public int height;

   public Size() {
      this.width = 0;
      this.height = 0;
   }

   public Size(int w, int h) {
      this.width = w;
      this.height = h;
   }

   public Size(Size s) {
      this.width = s.width;
      this.height = s.height;
   }

   public boolean equals(int w, int h) {
      return this.width == w && this.height == h;
   }

   public Size setWidth(int width) {
      this.width = width;
      return this;
   }

   public Size setHeight(int height) {
      this.height = height;
      return this;
   }

   public boolean isEmpty() {
      return this.width <= 0 || this.height <= 0;
   }

   public RectF toRectF() {
      float var2 = this.width;
      float var1 = this.height;
      return new RectF(0.0F, 0.0F, var2, var1);
   }

   public Size copy() {
      return new Size(this);
   }

   public String key() {
      return "w=" + this.width + "-h=" + this.height;
   }

   public Boolean isVertical() {
      return this.height > this.width;
   }

   public RectF ratioPageSize(Size maxPageSize) {
      if (Math.max(maxPageSize.width, maxPageSize.height) > Math.max(this.width, this.height)) {
         float var4 = this.width;
         float var8 = this.height;
         return new RectF(0.0F, 0.0F, var4, var8);
      }

      int var2;
      int var5;
      if (this.isVertical()) {
         var5 = maxPageSize.width;
         var2 = maxPageSize.height;
      } else {
         var5 = maxPageSize.height;
         var2 = maxPageSize.width;
      }

      float var6 = Math.max(this.width * 1.0F / var5, this.height * 1.0F / var2);
      float var3 = this.width / var6;
      float var7 = this.height / var6;
      return new RectF(0.0F, 0.0F, var3, var7);
   }

   @Override
   public boolean equals(Object o) {
      if (this == o) {
         return true;
      } else if (o != null && this.getClass() == o.getClass()) {
         o = (Size)o;
         return this.width == ((Size)o).width && this.height == ((Size)o).height;
      } else {
         return false;
      }
   }

   @Override
   public int hashCode() {
      return Objects.hash(this.width, this.height);
   }

   @NonNull
   @Override
   public String toString() {
      return "width:" + this.width + ",height:" + this.height;
   }
}

File: onyxsdk-base/support/onyxsdk-baselite/src/main/java/com/onyx/android/sdk/base/data/Size.kt
package com.onyx.android.sdk.base.data

import android.graphics.Rect
import android.graphics.RectF

data class Size(
    var width: Int = 0,
    var height: Int = 0,
) {
    fun setWidth(width: Int): Size {
        this.width = width
        return this
    }

    fun setHeight(height: Int): Size {
        this.height = height
        return this
    }

    fun isEmpty(): Boolean = width <= 0 || height <= 0

    fun equals(w: Int, h: Int): Boolean = width == w && height == h

    fun toRectF(): RectF = RectF(0.0f, 0.0f, width.toFloat(), height.toFloat())

    fun toRect(): Rect = Rect(0, 0, width, height)

    fun key(): String = "w=$width-h=$height"

    fun isVertical(): Boolean = height > width

    fun ratioPageSize(maxPageSize: Size): RectF {
        require(!isEmpty()) { "Size must have positive dimensions: $width x $height" }
        require(!maxPageSize.isEmpty()) {
            "Maximum page size must have positive dimensions: ${maxPageSize.width} x ${maxPageSize.height}"
        }
        val (maxWidth, maxHeight) = if (isVertical()) {
            maxPageSize.width to maxPageSize.height
        } else {
            maxPageSize.height to maxPageSize.width
        }
        val ratio = maxOf(
            1.0f,
            width.toFloat() / maxWidth.toFloat(),
            height.toFloat() / maxHeight.toFloat(),
        )
        return RectF(0.0f, 0.0f, width / ratio, height / ratio)
    }

    fun getOrElse(defaultValue: () -> Size): Size = if (isEmpty()) defaultValue() else this

    companion object {
        val empty = Size()

        fun android.util.Size.toDataSize(): Size = Size(width, height)
    }
}
